src/datasource_extensions.py

```diff
@@ -178,3 +250,100 @@ def download(self, clobber=False):
                 logger.info(
                     f"Downloaded and cropped {url} to {download_filepath}"
                 )
+
+    def _get_download_filepath(self, run_date, leadtime):
+        raw_dir = self._raw_base_dir / "daily"
+        raw_dir.mkdir(parents=True, exist_ok=True)
+        return (
+            raw_dir / f"chirpsgefs_{self._country_config.iso3}_"
+            f"{run_date.strftime('%Y-%m-%d')}_"
+            f"lt{str(leadtime).zfill(2)}d.tif"
+        )
+
+    def process(
+        self,
+        high_risk_hulls: gpd.GeoDataFrame,
+        bounds_buffer: float = 0.5,
+        resolution=0.01,
+    ):
+        # Loop through all dates
+        df_results_full = pd.DataFrame()
+        for run_date in self._date_range:
+            # Loop through all lead times
+            for leadtime in range(10):
+                df_output = self._process_single_file(
+                    run_date=run_date,
+                    leadtime=leadtime + 1,
+                    high_risk_hulls=high_risk_hulls,
+                    bounds_buffer=bounds_buffer,
+                    resolution=resolution,
+                )
+                if df_output is None:
+                    continue
+                df_results_full = pd.concat(
+                    (df_results_full, df_output), ignore_index=True
+                )
```

Comment:
I suppose we should adjust to include the day the forecast was made as we think it represents the next 24 hours?

Reply:
What do you mean?

Reply:
i think as soon as it is confirmed that the csv that is used by @PaulineNimo to analyze chirps-gefs is updated to reflect this - we should go ahead and merge!

Reply:
But I think there is no shift needed in the case of ECMWF, as leadtime 0 also has 0 rainfall

Reply:
yeah I agree don't need to shift on ECMWF. I thought I was pointing the `class ChirpsGefs(_DataSourceExtension)` chunk code. It could be adjust somewhere else rather than the datasource_extensions.py if it's better/easier?

Reply:
Yes I'll let @PaulineNimo take care of that in her own branch